sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,548 @@
+    ...         "overwrite_output_dir":True,
+    ...         "learning_rate":1e-4,
+    ...         "num_train_epochs":1,
+    ...         "per_device_train_batch_size":16,
+    ...     }
+    ... ) # doctest: +SKIP
+    >>> forecaster.fit(y = df, fh = [1,2,3,4,5]) # doctest: +SKIP
+    >>> y_pred = forecaster.predict() # doctest: +SKIP
+
+
+    >>> #Example fine-tuning with a pre-trained model
+    >>> from sktime.forecasting.hf_patchtst_forecaster import HFPatchTSTForecaster
+    >>> import pandas as pd
+    >>> dataset_path = pd.read_csv(
+    ...     "https://raw.githubusercontent.com/zhouhaoyi/ETDataset/main/ETT-small/ETTh1.csv"
+    ...     ).drop(columns = ["date"]
+    ... )
+    >>> from sklearn.preprocessing import StandardScaler
+    >>> scaler = StandardScaler()
+    >>> scaler.set_output(transform="pandas") # doctest: +SKIP
+    >>> scaler = scaler.fit(dataset_path.values) # doctest: +SKIP
+    >>> df = scaler.transform(dataset_path) # doctest: +SKIP
+    >>> df.columns = dataset_path.columns
+    >>> forecaster = HFPatchTSTForecaster(
+    ...     model_path="namctin/patchtst_etth1_forecast",
+    ...     fit_strategy = "minimal",
+    ...     training_args = {
+    ...         "output_dir":"/PatchTST/",
+    ...         "overwrite_output_dir":True,
+    ...         "learning_rate":1e-4,
+    ...         "num_train_epochs":1,
+    ...         "per_device_train_batch_size":16,
+    ...     }
+    ... ) # doctest: +SKIP
+    >>> forecaster.fit(y = df, fh = list(range(1,97))) # doctest: +SKIP
+    >>> y_pred = forecaster.predict() # doctest: +SKIP
+    """
+
+    _tags = {
+        "X_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "scitype:y": "both",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": False,
+        "capability:pred_int": False,
+        "capability:pred_int:insample": False,
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "python_dependencies": ["transformers", "torch"],
+        "capability:global_forecasting": True,
+    }
+
+    def __init__(
+        self,
+        # model variables except for forecast_columns
+        model_path=None,
+        fit_strategy="full",
+        validation_split=0.2,
+        config=None,
+        training_args=None,
+        compute_metrics=None,
+        callbacks=None,
+    ):
+        self.model_path = model_path
+        self.fit_strategy = fit_strategy
+        # dataset and training parameters
+        self.validation_split = validation_split
+        self.config = config
+        self.training_args = training_args
```

Comment:
use `self._training_args` to create `{}` instead of `None`
this currently fails if no `training_args` is provided during model init